app/models/grouping.rb

```diff
@@ -524,6 +531,14 @@ def past_collection_date?
     collection_date < Time.current
   end
 
+  # Return the duration of this grouping's assignment or the time between now and the collection_date whichever is
+  # less. If the collection date has already passed, return 0 seconds.
+  def adjusted_duration
+    add = assignment.submission_rule.periods.pluck(:hours).sum.hours + (extension&.time_delta || 0)
```

Comment:
I'm thinking the submission rule periods shouldn't be counted as part of the reported duration, but rather listed separately.

Reply:
Ok I see that. I think we should discuss separately exactly what info we should give the students in their view. That will inform what these various methods do.